eventiq_asb/broker.py

```diff
@@ -192,7 +194,8 @@ async def ack(self, raw_message: ServiceBusReceivedMessage) -> None:
         receiver = self._receivers.pop(
             id(raw_message), None
         )  # retrieve receiver reference for given message
-        if receiver:
+        # check if raw message is already settled else error is raised - private method
+        if receiver and not raw_message._settled:  # noqa: SLF001
```

Comment:
jaki jest case że message jest już `settled`? Rozumiem że w metodzie `receiver.complete_message()` ten atrybut nie jest sprawdzany i się wywali? Czy można w takim razie zrobić `abandon_message` na message który jest `_settled` ?

Reply:
w glownym eventiq jesli leci fail / skip to wiadomosc jest ack. Przy czym w ASB taka wiadomosc jest juz settled

Reply:
uzupelniajac. W momencie kiedy jest fail / skip w _handle_message_finalization zostaje odpalony dispatch_after. 
W DLQ middleware wiadomosc zostaje przeniesiona do dlq i wtedy jest juz settled